### PR TITLE
Resources del 1651

### PR DESCRIPTION
**Pull request by KirbyKatcher**

### CODE REVIEW

**Comment by galasa-team** [comment]
Build successful

File: galasa-parent/dev.galasa.framework.api.common/src/main/java/dev/galasa/framework/api/common/resources/CPSProperty.java [hunk 1]
```diff
@@ -73,18 +73,18 @@ public String getValue() {
 
     public String getOutputValue() {
         String outputValue  = this.value;
-        if(namespace.getVisibility() == Visibility.SECURE){
+        if (namespace.getVisibility() == Visibility.SECURE){
```
Comment (techcobweb) [inline galasa-parent/dev.galasa.framework.api.common/src/main/java/dev/galasa/framework/api/common/resources/CPSProperty.java:76]: I would have added a namespace.isSecure() boolean method.

File: galasa-parent/dev.galasa.framework.api.common/src/main/java/dev/galasa/framework/api/common/resources/CPSNamespace.java [hunk 2]
```diff
@@ -81,7 +81,7 @@ public CPSProperty getPropertyFromStore(String propertyName) throws Configuratio
         if (visibility != Visibility.HIDDEN) {
             GalasaPropertyName propName = new GalasaPropertyName(this.name, propertyName);
             prop = new CPSProperty(this.propertyStore, this, propName);
-            prop.LoadValueFromStore();
+            prop.loadValueFromStore();
```
Comment (techcobweb) [inline galasa-parent/dev.galasa.framework.api.common/src/main/java/dev/galasa/framework/api/common/resources/CPSNamespace.java:84]: Why is this a public method ?

And 
```
public CPSProperty getProperty(String propertyName) throws ConfigurationPropertyStoreException {
        CPSProperty prop = getPropertyFromStore(propertyName);
            if (!prop.existsInStore()) {
                prop = null;
            }
            return prop;
        }
```
Why do we need that ? Why can't you create a property which doesn't exist in the store yet, and then tell the property to load/save ?

File: galasa-parent/dev.galasa.framework.api.common/src/main/java/dev/galasa/framework/api/common/resources/CPSProperty.java (hunk 1 above)
Comment (techcobweb) [reply galasa-parent/dev.galasa.framework.api.common/src/main/java/dev/galasa/framework/api/common/resources/CPSProperty.java:76]: lol. You already have that method, so use it instead ?

File: galasa-parent/dev.galasa.framework.api.common/src/main/java/dev/galasa/framework/api/common/resources/CPSNamespace.java (hunk 2 above)
Comment (techcobweb) [reply galasa-parent/dev.galasa.framework.api.common/src/main/java/dev/galasa/framework/api/common/resources/CPSNamespace.java:84]: Don't think I like this method. Not easy to see what's using it.

File: galasa-parent/dev.galasa.framework.api.resources/src/main/java/dev/galasa/framework/api/resources/routes/ResourcesRoute.java [hunk 3]
```diff
@@ -95,54 +95,61 @@ public void processDataArray(JsonArray jsonArray, String action) throws Internal
...
             String apiversion = resource.get("apiVersion").getAsString();
             String expectedApiVersion = GalasaProperty.DEFAULTAPIVERSION;
-            if (apiversion.equals(expectedApiVersion)){
+            if (apiversion.equals(expectedApiVersion)) {
                 GalasaProperty galasaProperty = gson.fromJson(resource, GalasaProperty.class);           
-                if (galasaProperty.isPropertyValid()){
+                if (galasaProperty.isPropertyValid()) {
                     CPSFacade cps = new CPSFacade(framework);
                     CPSNamespace namespace = cps.getNamespace(galasaProperty.getNamespace());
                     CPSProperty property = namespace.getPropertyFromStore(galasaProperty.getName());
```
Comment (techcobweb) [inline galasa-parent/dev.galasa.framework.api.resources/src/main/java/dev/galasa/framework/api/resources/routes/ResourcesRoute.java:116]: Won't this be null now, if the property isn't in the store ?

File: galasa-parent/dev.galasa.framework.api.resources/src/main/java/dev/galasa/framework/api/resources/routes/ResourcesRoute.java [hunk 4]
```diff
@@ -95,54 +95,61 @@ public void processDataArray(JsonArray jsonArray, String action) throws Internal
...
-            }else{
+            } else {
                 ServletError error = new ServletError(GAL5027_UNSUPPORTED_API_VERSION, apiversion, expectedApiVersion);
                 throw new InternalServletException(error, HttpServletResponse.SC_BAD_REQUEST);
             }
-        }catch (InternalServletException i){
+        } catch (InternalServletException i){
             throw i;
-        }catch (NullPointerException n){
+        } catch (NullPointerException n){
```
Comment (techcobweb) [inline galasa-parent/dev.galasa.framework.api.resources/src/main/java/dev/galasa/framework/api/resources/routes/ResourcesRoute.java:146]: We need a way of avoiding catching these surely. Catching NPEs is a recipe for disaster.

File: galasa-parent/dev.galasa.framework/src/main/java/dev/galasa/framework/spi/utils/GalasaGsonBuilder.java [hunk 5]
```diff
@@ -13,7 +13,7 @@
 public class GalasaGsonBuilder {
 
     public static Gson build() {
-        GsonBuilder builder = new GsonBuilder();
+        GsonBuilder builder = new GsonBuilder().disableHtmlEscaping();
```
Comment (techcobweb) [inline galasa-parent/dev.galasa.framework/src/main/java/dev/galasa/framework/spi/utils/GalasaGsonBuilder.java:16]: I think we need to explain why we have to disable html escaping. It's not clear why we need to do this.

**Review by techcobweb — APPROVED** [review]
Some comments from me which need resolving, but it's getting there.

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-framework-pr-480-4lgwh for details. If you are unable to do so, please contact a member of the Galasa team.

File: galasa-parent/dev.galasa.framework.api.resources/src/main/java/dev/galasa/framework/api/resources/routes/ResourcesRoute.java (hunk 3 above)
Comment (KirbyKatcher) [reply galasa-parent/dev.galasa.framework.api.resources/src/main/java/dev/galasa/framework/api/resources/routes/ResourcesRoute.java:149]: That is exactly what it is doing the only time it does not do that is if the namespace is hidden to force an exception

**Comment by galasa-team** [comment]
Build failed, see http://localhost:8001/api/v1/namespaces/tekton-pipelines/services/tekton-dashboard:http/proxy/#/namespaces/galasa-build/pipelineruns/repo-framework-pr-480-dp2sp for details. If you are unable to do so, please contact a member of the Galasa team.

**Comment by galasa-team** [comment]
Build successful

**Comment by galasa-team** [comment]
Build successful

**Review by techcobweb — APPROVED** [review]

**Pull request merged**